docs/specifications/tests/Nameserver-TP/nameserver14.md

```diff
@@ -0,0 +1,87 @@
+# NAMESERVER14: Test for unknown version with unknown OPTION-CODE
+
+## Test case identifier
+
+**NAMESERVER14** 
+
+### Objective
+
+This test case actually combines the test options in test cases
+[NAMESERVER10](../nameserver10.md) and [NAMESERVER11](../nameserver11.md). 
+
+### Inputs
+
+"Child Zone" - The domain name to be tested.
+
+### Ordered description of steps to be taken to execute the test case
+
+1. Create an SOA query for the *Child Zone* with an OPT record with 
+   EDNS version set to "1" and  with EDNS OPTION-CODE set to an
+   unassigned value (see [IANA-DNSSYSTEM-PARAMETERS]) and
+   with no other EDNS options or flags. 
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+3. For each name server in *Name Server IP* do:
+
+	1. Send the SOA query to the name server and collect the response.
+	2. If there is no DNS response, output *[NO_RESPONSE]* and go to
+      	next server.
+	3. Else, if the DNS response has the RCODE "FORMERR" then output
+      	*[NO_EDNS_SUPPORT]*.
+	4. Else, if the RCODE has "NOERROR", and has an "OPTION-CODE" and
+	"Version Number" present in the response,  then output
+	[UNKNOWN_OPTION_CODE_VERSION]. 
+	5. Else, if the DNS response has the RCODE "NOERROR" and "Version
+	number" present in the response then output [UNSUPPORTED_EDNS_VER]
+	6. Else, if the DNS response has the RCODE "NOERROR" and "OPTION-CODE"
+	present in the response then output [UNKNOWN_OPTION_CODE].
```

Comment:
I guess this is not needed. The rest of the errors go to NS_ERROR.

BTW, it is not possible to have an OPT record without ENDS version is set.

Reply:
@matsduf i am following this : 

EDNS - Unknown Version with Unknown Option
dig +norec +edns=100 +noednsneg +ednsopt=100 soa zone @server
expect: BADVERS
expect: OPT record with version set to 0
expect: not to see SOA
expect: that the option will not be present in response
See RFC6891

Reply:
The problem is not to define good behavior. The problem is to label/classify everything else.

Reply:
I do not understand your statement. EDNS compliance test follows all these tests : http://users.isc.org/~marka/summary.html

Reply:
@sandoche2k, the ISC link specifies what to expect when everything is compliant, and then the following step is fine,
```
   7. Else, if the DNS response meet the following four criteria, then just go to the 
      next name server (no error):
      1. If the SOA is not obtained as response in the ANSWER section.
      2. It has the RCODE "BADVERS".
      3. The pseudo-section response has an OPT record with version set to 0.
      4. The option is not present in the response
```
We have identified two clear "error" cases:
```
   2. If there is no DNS response, output *[NO_RESPONSE]* and go to
      next server.
   3. Else, if the DNS response has the RCODE "FORMERR" then output
      *[NO_EDNS_SUPPORT]*.
```
The problem is what we should output in other cases. We have responses with NOERROR with and without OPT record. We have responses with OPT record with and without OPTION-CODE. And those responses with OPT record can have EDNS version 0 or other version. (An OPT record always has an EDNS version.)